corehq/form_processor/models.py

```diff
@@ -524,6 +527,31 @@ def transactions(self):
     def case_attachments(self):
         return {attachment.name: attachment for attachment in self.get_attachments()}
 
+    def modified_since_sync(self, sync_log):
+        if self.server_modified_on >= sync_log.date:
+            # check all of the transactions since last sync for one that had a different sync token
+            from corehq.form_processor.backends.sql.dbaccessors import CaseAccessorSQL
+            return CaseAccessorSQL.case_has_transactions_since_sync(self.case_id, sync_log._id, sync_log.date)
+            # TODO: Should this logic really be on CaseAccessorSQL? I'm following the pattern of the
+            #       transactions method.
+        return False
+
+    def get_actions_for_form(self, xform):
+        from casexml.apps.case.xform import get_case_updates
+        # Will this import cause atrocious performance?
```

Comment:
This should be OK. It gets called once per case in the form (which is usually a small number). This is another thing that could be optimized though since its basically doing exactly the same work for each case (since the form will be the same). Fine for now though.